Modules/Core/Common/include/itkArray.h

```diff
@@ -154,67 +171,73 @@ class ITK_TEMPLATE_EXPORT Array : public vnl_vector< TValue >
    * the responsibility of freeing the memory for this data.  If
    * "LetArrayManageMemory" is true, then this class will free the
    * memory when this object is destroyed. */
-  void SetData(TValue *data, SizeValueType sz, bool LetArrayManageMemory = false);
+  void
+  SetData( TValue * data, SizeValueType sz, bool LetArrayManageMemory = false );
 
 #ifdef __INTEL_COMPILER
-#pragma warning disable 444 //destructor for base class "itk::Array<>" is not virtual
+#  pragma warning disable 444 // destructor for base class "itk::Array<>" is not virtual
 #endif
   /** This destructor is not virtual for performance reasons. However, this
    * means that subclasses cannot allocate memory. */
   ~Array();
 
-#if ! defined ( ITK_LEGACY_REMOVE )
-  void swap(Array &other)
-    {
-      this->Swap(other);
-    }
+#if !defined( ITK_LEGACY_REMOVE )
+  void
+  swap( Array & other )
+  {
+    this->Swap( other );
+  }
 #endif
 
-  void Swap(Array &other)
-    {
-      using std::swap;
-      this->VnlVectorType::swap(other);
-      swap(this->m_LetArrayManageMemory, other.m_LetArrayManageMemory);
-    }
+  void
+  Swap( Array & other )
+  {
+    using std::swap;
+    this->VnlVectorType::swap( other );
+    swap( this->m_LetArrayManageMemory, other.m_LetArrayManageMemory );
+  }
 
 private:
-
   bool m_LetArrayManageMemory;
 };
 
-template< typename TValue >
-std::ostream & operator<<(std::ostream & os, const Array< TValue > & arr)
+template < typename TValue >
+std::ostream &
+operator<<( std::ostream & os, const Array< TValue > & arr )
 {
   os << "[";
   const unsigned int length = arr.size();
   if ( length >= 1 )
-    {
-    const unsigned int   last   = length - 1;
+  {
+    const unsigned int last = length - 1;
     for ( unsigned int i = 0; i < last; ++i )
-      {
+    {
       os << arr[i] << ", ";
-      }
-      os << arr[last];
     }
+    os << arr[last];
+  }
   os << "]";
   return os;
 }
 
 // declaration of specialization
-template<> ITKCommon_EXPORT std::ostream & operator<< <double> (std::ostream & os, const Array< double > & arr);
-template<> ITKCommon_EXPORT std::ostream & operator<< <float> (std::ostream & os, const Array< float > & arr);
+template <>
+ITKCommon_EXPORT std::ostream & operator<<< double >( std::ostream & os, const Array< double > & arr );
```

Comment:
I guess we plan to manually override `clang-format` in instances like this?

Reply:
Why?  

AlwaysBreakTemplateDeclarations = Yes .  -- Perhaps we want "MultiLine"

Reply:
`operator<< <double>` is clearer than `operator<<< double >`, at least to me. At first, the formatted version was confusing - is there an `<<<` operator?